sql/moz-fx-data-shared-prod/monitoring_derived/bigquery_usage_v2/query.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+
+"""
+Determine BigQuery usage.
+
+To read more on the source table, please visit:
+https://cloud.google.com/bigquery/docs/information-schema-jobs-by-organization
+"""
+
+from argparse import ArgumentParser
+
+from google.cloud import bigquery
+
+DEFAULT_PROJECTS = [
+    "moz-fx-data-shared-prod",
+]
+
+parser = ArgumentParser(description=__doc__)
+parser.add_argument("--date", required=True)  # expect string with format yyyy-mm-dd
+parser.add_argument("--project", default="moz-fx-data-shared-prod")
+# projects queries were run from that access table
+parser.add_argument("--source_projects", nargs="+", default=DEFAULT_PROJECTS)
+parser.add_argument("--destination_dataset", default="monitoring_derived")
+parser.add_argument("--destination_table", default="bigquery_usage_v2")
+
+# total_bytes_billed a minimum of 10MB per table total_bytes_processed
+
+
+def create_query(date, source_project):
+    """Create query for a source project."""
+    return f"""
+        SELECT
+         "{source_project}" AS source_project,
+          DATE('{date}') AS creation_date,
+          job_id,
+          job_type,
+          reservation_id,
+          cache_hit,
+          state,
+          referenced_tables.project_id AS reference_project_id,
+          dataset_id AS reference_dataset_id,
+          table_id AS reference_table_id,
+          destination_table.project_id AS  destination_project_id,
+          destination_table.dataset_id AS  destination_dataset_id,
+          destination_table.table_id AS  destination_table_id,
+          user_email,
+          REGEXP_EXTRACT(query, r"Username: (.*?),") AS username,
```

Comment:
There's a bit of an unfortunate complication here, which is that [JOBS_BY_ORGANIZATION](https://cloud.google.com/bigquery/docs/information-schema-jobs-by-organization) doesn't actually have the `query` column. Even though it's listed, its description says `SQL query text. Only the JOBS_BY_PROJECT view has the query column.`.

It's possible to use `JOBS_BY_ORGANIZATION` to get job ids and projects, then get associated query information from e.g. ``[PROJECT].`region-us`.INFORMATION_SCHEMA.JOBS_BY_PROJECT``, but that's going to be more complicated. Airflow has access to this metadata for all projects the same way it has access to `JOBS_BY_ORGANIZATION`.

Aside: I'm guessing the parsing of `query` here is using either Looker or STMO comment metadata. This definitely isn't going to be valid for all queries, but that's probably known for `bigquery_usage_v1`.

Reply:
Thanks for the detailed feedback!  I have added the appropriate codeowners entry.  I will leave out the parsing of the`query` field for now.